### PR TITLE
feat: auth setup

### PR DESCRIPTION
**Pull request by Karol-Wisniewski**

### CODE REVIEW

File: client/src/components/custom/login-dialog/login-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+  function onSubmit(values: UserCredentials) {
+    setIsLoading(true);
+    login(values)
+      .then(() => {
+        closeLoginModal();
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        console.error(error);
```
Comment (amordzon) [inline client/src/components/custom/login-dialog/login-dialog.tsx:52]: should we leave this?

File: client/src/components/custom/register-dialog/register-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  function onSubmit(values: UserRegisterData) {
+    setIsLoading(true);
+    registerUser(values)
+      .then(() => {
+        closeRegisterModal();
+        setIsLoading(false);
+        toast.success('Account created successfully');
+      })
+      .catch((error) => {
+        console.error(error);
```
Comment (amordzon) [inline client/src/components/custom/register-dialog/register-dialog.tsx:53]: hmmm

**Review by amordzon — APPROVED** [review]
nice

File: client/src/components/custom/login-dialog/login-dialog.tsx (hunk 1 above)
Comment (Karol-Wisniewski) [reply client/src/components/custom/login-dialog/login-dialog.tsx:52]: At this stage of app development, I think we should

**Pull request merged**